go/tools/gazelle/packages/walk.go

```diff
@@ -28,6 +29,14 @@ import (
 	"strings"
 )
 
+// packageReader holds the state used to read package metadata from a directory.
+// This mostly exists to avoid passing the same of parameters to every function.
```

Comment:
I've changed the comments to indicate this. Also moved the definition below `WalkFunc` and `Walk`, closer to the methods.